wordpress-mobile/WordPress-Android
# Redesign Jetpack Landing Screen - with looping text animation

Closes #17073 +
Closes #17074 too since it adds the default infinite looping animation to the large text.

This PR implements the Jetpack landing screen redesign with Compose and adds the infinite looping animation to the large text in the background.

A couple of remarks:
- The looping text animation is done with a `LazyColumn`, though it may need to be changed in the future when we'll influence the scrolling direction and speed with the gyro sensor data.
- The frosted glass effect with blur is only working on versions of Android 12 and above, while on earlier Android versions the text behind the buttons container is semi-transparent to make the buttons stand out.
- This PR doesn't add events tracking, that is to be handled later.
- The UI wasn't optimised for tablets.

To test:

<details>
<summary><b>Prerequisite</b> - Enable the <code>LandingScreenRevamp</code> feature flag</summary>

- Either `login` → `Me` → `App Settings` → `Debug Settings` → enable `LandingScreenRevamp` → `RESTART THE APP` → `logout` or
- Apply this hard-coded solution:

<details>
<summary>Patch to enable the <code>LandingScreenRevamp</code> feature flag</summary>

```diff
diff --git a/WordPress/build.gradle b/WordPress/build.gradle
index 63260a4ca8..fe999fa92f 100644
--- a/WordPress/build.gradle
+++ b/WordPress/build.gradle
@@ -112,7 +112,7 @@ android {
         buildConfigField "boolean", "READER_COMMENTS_MODERATION", "false"
         buildConfigField "boolean", "SITE_INTENT_QUESTION", "true"
         buildConfigField "boolean", "SITE_NAME", "false"
-        buildConfigField "boolean", "LANDING_SCREEN_REVAMP", "false"
+        buildConfigField "boolean", "LANDING_SCREEN_REVAMP", "true"
         buildConfigField "boolean", "LAND_ON_THE_EDITOR", "false"
         buildConfigField "boolean", "BLOGGING_PROMPTS", "false"
         buildConfigField "boolean", "QUICK_START_EXISTING_USERS_V2", "false"
```  
</details>

---

</details>

##### 1️⃣ Test Case 1: `Android12+` — Blurred text behind the buttons

0. Make sure `LandingScreenRevamp` feature flag is enabled (see prerequisite) and you're not logged in
1. Launch Jetpack App on a device or emulator with Android12 or higher
   **Verify** That the design implementation matches the design specs and the buttons container is blurring the text.
3. Let the text animation play until you see the same text scrolling at least 3 times including the initial show.
   **Expect** A smooth scrolling animation (there should be no jumps or jiggles or anything non-smooth).
4. Toggle dark mode on the device.
   **Expect** The design of the screen to adjust for night.
5. Rotate the screen to landscape
   **Expect** The design of the screen to include the Jetpack logo and the buttons to not be cut.
6. Set larger and smaller font sizes via Accessibility settings
   **Expect** The animation to still run smooth and the buttons to be visible and their text readable.
7. Try to scroll the text up and down with touch gestures
   **Expect** The scrolling animation to continue, ignoring the user scroll.

##### 2️⃣ Test Case 2: `Android12-` — Semi-transparent text behind the buttons

0. Make sure `LandingScreenRevamp` feature flag is enabled (see prerequisite) and you're not logged in
1. Launch Jetpack App on a device or emulator with Android11 or lower
   **Expect** The large text behind the buttons container to be semi-transparent and not blurred.

##### 3️⃣ Test Case 3: `Login/Signup` — Functionality

0. Make sure `LandingScreenRevamp` feature flag is enabled (see prerequisite) and you're not logged in
1. Launch Jetpack App
2. Tap the `Continue with WordPress.com` button
   **Expect** The WP.com login/signup flow to start and work as before.
   - Continue the flow expecting it to work as before.
   - Logout.
3. Tap the `Enter your existing site address` button
   **Expect** The login with WordPress site to work as before.
   - Continue the flow expecting it to work as before.

##### 4️⃣ Test Case 4: `WordPress` app — Regression test
0. Make sure `LandingScreenRevamp` feature flag is enabled (see prerequisite) and you're not logged in
1. Launch the WordPress app and
   **Expect** The landing screen to be unchanged.

---

## Previews

**Remark** The framerate of the GIF preview was cut to 10fps to be able to upload them here. The animation should run smoother on the device.

| Android 12+ Light | Android 12+ Dark |
| --- | --- |
| ![ezgif-1-d2bd823dff](https://user-images.githubusercontent.com/4588074/188169837-e830ea13-9f8f-4a0e-8c8c-7ade64e26153.gif) | ![ezgif-1-41db605664](https://user-images.githubusercontent.com/4588074/188169722-c6338cea-9c48-47fd-8c3a-a440f3dc89ae.gif) | 

| Android 12- Light | Android 12- Dark |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/4588074/188171082-9562fc7c-85f7-4325-b9e9-1860cd13053d.gif" alt="preview_a12-_light" width="320"> | <img src="https://user-images.githubusercontent.com/4588074/188172103-bb2e3a0e-91b9-4f5a-a6b6-cc873db4243d.gif" alt="preview_a12-_dark" width="320"> |

<details>
<summary>🏞 Landscape Previews on <code>Android12+</code></summary>

| Light Mode |
| --- |
| ![preview_landscape_light](https://user-images.githubusercontent.com/4588074/188172288-857ec6a8-e08d-4bc7-b81a-e5cfbec16f6c.png) |

| Dark Mode |
| --- |
| ![preview_landscape_dark](https://user-images.githubusercontent.com/4588074/188172273-2223194a-221e-4f95-bf91-0d36b0bc328b.png) |

</details>

<details>
<summary>🎬  Smooth Recording Preview on <code>Android12+</code> in <code>Dark</code> mode</summary>

https://user-images.githubusercontent.com/4588074/188173335-55e4290d-744f-452f-8f58-0dd3f315e736.mp4

</details>

<details>
<summary>🎬 Smooth Recording Preview on <code>Android12-</code> in <code>Light</code> mode</summary>

https://user-images.githubusercontent.com/4588074/188173814-b8a272e5-7afe-438d-b0dd-2a8240d19255.mp4

</details>

<details>
<summary>🖥 Tablet Previews</summary>

| Light Mode |
| --- |
| ![tablet_light](https://user-images.githubusercontent.com/4588074/188175865-1f4ec4d0-637f-4bb8-9266-d37c1339033a.png)

| Dark Mode |
| --- |
| ![tablet_night](https://user-images.githubusercontent.com/4588074/188175890-379f94cc-7283-479f-8f21-14343d5e7502.png) |

</details>

### Merging Strategy

1. Merge #17104 into `feature/revamp-landing-screen/`
2. Manually fix the merge conflicts of this branch by merging it locally with `feature/revamp-landing-screen/` and pushing
3. Merge this branch to `feature/revamp-landing-screen`
4. Open a PR to merge `feature/revamp-landing-screen` to `trunk`

## Regression Notes
5. Potential unintended areas of impact
   Unexpected, since the changes introduce a new screen.
   The `LoginActivity` was changed though.

8. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing.

9. What automated tests I added (or what prevented me from doing so)
   This PR only adds a new UI, these changes don't warrant tests according to our current guidelines.

PR submission checklist:
